### PR TITLE
WMS Reverse Proxy Fix

### PR DESCRIPTION
**Pull request by tst-jzmolek**
This is a potential fix for an issue that Max has been seeing using reverse proxies. I am waiting for his feedback from his testing, but thought I'd put the PR out there to view as well.

I have tested this on CentOS7 with ATAK and QGIS clients.

To set up a reverse proxy:

1. Install Apache 2.4 on a separate machine from GEE (non-bundled with GEE since the GEE-specific version doesn't include all the reverse proxy capabilities)
2. Modify httpd.conf to set it up as a proxy:
```
LoadModule proxy_module modules/mod_proxy.so
LoadModule proxy_fcgi_module modules/mod_proxy_fcgi.so
LoadModule proxy_scgi_module modules/mod_proxy_scgi.so
LoadModule proxy_fdpass_module modules/mod_proxy_fdpass.so
LoadModule proxy_wstunnel_module modules/mod_proxy_wstunnel.so
LoadModule proxy_ajp_module modules/mod_proxy_ajp.so
LoadModule proxy_express_module modules/mod_proxy_express.so
LoadModule proxy_http_module modules/mod_proxy_http.so
LoadModule ssl_module modules/mod_ssl.so

# To proxy on the http port use this, otherwise modify ssl.conf for https
ProxyPass "/" "youractualgeeserver.net"
ProxyPassReverse"/" "youractualgeeserver.net" # Probably not needed
```

3. If testing https, modify ssl.conf
```
SSLEngine on

SSLProxyEngine on
# For my simple testing I disabled some checks
SSLProxyVerify none
SSLProxyCheckPeerCN off
SSLProxyCheckPeerName off

ProxyPass "/" "https://youractualgeeserver.net:443/"
ProxyPassReverse "/" "https://youractualgeeserver.net:443/"

# I also created my own certificate file and updated SSLCertificateFile and SSLCertificateKeyFile, but that probably wasn't needed since it came with some defaults already
```
4. Restart the proxy server: `sudo systemctl restart httpd`

### CODE REVIEW

File: earth_enterprise/src/server/wsgi/wms/ogc/common/utils.py [hunk 1]
```diff
@@ -137,7 +141,14 @@ def GetServerURL(environ):
   complete_url += urllib.quote(environ.get("REDIRECT_URL", ""))
   complete_url += urllib.quote(environ.get("PATH_INFO", ""))
 
-  return server_url, complete_url
+  if environ.get("HTTP_X_FORWARDED_HOST"):
```
Comment (tst-dsugar) [inline earth_enterprise/src/server/wsgi/wms/ogc/common/utils.py]: What if it is https?

Comment (tst-jzmolek) [reply earth_enterprise/src/server/wsgi/wms/ogc/common/utils.py]: I tested both. It still uses this in the header either way.

**Comment by tst-blaird** [comment]
@tst-jzmolek do you have steps for setting up a proxy to test this?

**Comment by tst-jzmolek** [comment]
> @tst-jzmolek do you have steps for setting up a proxy to test this?

Sure. I had to do a separate install of Apache 2.4 (non-GEE bundled) on a different system to use as the reverse proxy. Our bundled version doesn't build everything to do reverse proxying. Then set up the config to do reverse proxying to the actual GEE server. I'll put more details back up in the main description.

**Comment by tst-jzmolek** [comment]
Yes, the description is now updated with these steps.

**Comment by tst-blaird** [comment]
@tst-jzmolek After retargeting the PR to master there are additional files in the changed list.  Are these changes ok to merge into master?

**Comment by tst-jzmolek** [comment]
I don't think all of them, no, There's a reference to 5.3.3 that I think should stay master, so I'll have to clean that up.

**Pull request closed without merging**